ddtrace/internal/core/__init__.py

```diff
@@ -314,21 +284,15 @@ def add_suppress_exception(exc_type: type) -> None:
     _CURRENT_CONTEXT.get()._suppress_exceptions.append(exc_type)
 
 
-def get_item(data_key: str, span: Optional["Span"] = None) -> Any:
-    _deprecate_span_kwarg(span)
-    if span is not None and span._local_root is not None:
-        return span._local_root._get_ctx_item(data_key)
```

Comment:
My guess about why this change broke a lot of tests is that there's a difference in lifespan between `span._local_root` and the span's corresponding `ExecutionContext` in some scenarios. I think that if these `get_`/`set_` functions used `span._get_ctx_item` instead of `span._local_root._get_ctx_item` the removal would be simpler.